client/src/main/java/org/eclipse/hono/client/CommandClient.java

```diff
@@ -0,0 +1,33 @@
+package org.eclipse.hono.client;
+
+import io.vertx.core.Future;
+import io.vertx.core.buffer.Buffer;
+
+/**
+ * A client for accessing Hono's Command and Control API.
+ * <p>
+ * An instance of this interface is always scoped to a specific tenant and device.
+ * </p>
+ */
+public interface CommandClient extends RequestResponseClient {
+
+    /**
+     * Sends a command to a device and expects a response.
+     * <p>
+     * A device needs to be (successfully) registered before a client can upload
+     * any data for it. The device also needs to be connected for a successful delivery.
+     *
+     * @param command The command name.
+     * @param data The command data to send to the device.
+     * @return A future indicating the result of the operation.
+     *         <p>
+     *         The future will succeed if a response with status 2xx has been received from the device.
+     *         <p>
```

Comment:
will the future be completed with `null` or with an empty buffer if the response has no payload?